mesh/mesh.go

```diff
@@ -500,3 +500,66 @@ func (m *Mesh) GetATXs(atxIds []types.AtxId) (map[types.AtxId]*types.ActivationT
 	}
 	return atxs, mIds
 }
+
+// ActiveSetForLayerView - returns the active set size that matches the view of the contextually valid blocks in the provided layer
+func (m *Mesh) ActiveSetForLayerView(layer types.LayerID, layersPerEpoch uint16) (uint32, error) {
+
+	epoch := layer.GetEpoch(layersPerEpoch)
+	firstLayerOfPrevEpoch := types.LayerID(epoch-1) * types.LayerID(layersPerEpoch)
+
+	// build a map of all blocks between firstLayerOfPrevEpoch and our current layer
+	mp := make(map[types.BlockID]struct{})
+	blocks, err := m.LayerBlockIds(layer)
+	for _, bid := range blocks {
+		mp[bid] = struct{}{}
+	}
+
+	activeSetSize := uint32(0)
+	countedAtxs := make(map[string]types.AtxId)
+
+	traversalFunc := func(blkh *types.BlockHeader) error {
+
+		// only contextually valid blocks
+		if !m.IsContexuallyValid(blkh.Id) {
```

Comment:
this is wrong, traversing the consensus view means that you choose only the contextual blocks in the starting layer and traverse them. While traversing you consider all syntactically valid blocks, not only the contextually valid blocks. Call me later and we'll discuss that point in more details

Reply:
you ignored this comment...

Reply:
Fixed